src/inference/include/openvino/runtime/intel_npu/properties.hpp

```diff
@@ -72,5 +72,20 @@ static constexpr ov::Property<std::string> compilation_mode_params{"NPU_COMPILAT
  */
 static constexpr ov::Property<bool> turbo{"NPU_TURBO"};
 
+/**
+ * @brief [Only for NPU Compiler]
+ * Type: integer, default is -1
+ * Sets the number of npu tiles to compile the model for.
+ */
+static constexpr ov::Property<int64_t> tiles{"NPU_TILES"};
+
+/**
+ * @brief
+ * Type: integer, default is -1
+ * Maximum number of tiles supported by the device we compile for. Can be set for offline compilation. If not set, it
```

Comment:
Compiler always needs NPU_MAX_TILES today either to select proper default values depending on PERFORMANCE_MODE or to compare NPU_TILES against it. Compiler would throw an exception in case NPU_TILES > NPU_MAX_TILES.
For online compilation or offline compilation with CID, NPU_TILES is enough. Plugin would query the available number of tiles from the driver and populate NPU_MAX_TILES.
For offline compilation with CIP we currently need to pass NPU_MAX_TILES as well from the application ( this behaviour is not changed by the current PR).